test/e2e/setstate_atomicity.cpp

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2019-2020 Xenios SEZC
+// https://www.veriblock.org
+// Distributed under the MIT software license, see the accompanying
+// file LICENSE or http://www.opensource.org/licenses/mit-license.php.
+
+#include <gtest/gtest.h>
+
+#include <util/pop_test_fixture.hpp>
+
+using namespace altintegration;
+
+/**
+ * Mine chain A: endorse 2 valid altchain blocks
+ * Mine chain B: endorse 1 block, add an invalid VTB to this block
+ * Activate chain A
+ * Attempt to activate chain B
+ * Check that the chain B activation attempt failed and left the state unchanged
+ */
+
+struct SetStateAtomicity : public ::testing::Test, public PopTestFixture {};
+
+TEST_F(SetStateAtomicity, setStateAtomicity) {
+  AltTree testAlttree = AltTree(altparam, vbkparam, btcparam);
```

Comment:
you could use `this->alttree` which is defined and bootstrapped in PopTestFixture